components/net_traits/bluetooth_blacklist.rs

```diff
@@ -0,0 +1,120 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use regex::Regex;
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::Path;
+use std::string::String;
+
+const NOT_VALID_UUID: &'static str = "This UUID is not valid";
+const BLACKLIST_PARSE_ERROR: &'static str = "The filling of the blacklist resulted in error";
+const EXCLUDE_READS: &'static str = "exclude-reads";
+const EXCLUDE_WRITES: &'static str = "exclude-writes";
+const VALID_UUID_REGEX: &'static str = "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}";
+const FILE_PATH: &'static str = "./resources/blacklist.txt";
+const EXCLUDE_ERR: &'static str = "This UUID is blacklisted";
+const EXCLUDE_READS_ERR: &'static str = "This UUID is blacklisted for reads";
+const EXCLUDE_WRITES_ERR: &'static str = "This UUID is blacklisted for writes";
+
+pub struct BluetoothBlacklist {
+    blacklist: HashMap<String, ValueType>,
+    error_string: String,
+}
+
+#[derive(Eq, PartialEq)]
+pub enum ValueType {
+    Exclude,
+    ExcludeReads,
+    ExcludeWrites,
+}
+
+impl BluetoothBlacklist {
+    pub fn new() -> BluetoothBlacklist {
```

Comment:
if you don't use the `new` anywhere else, than this shouldn't be `pub`. (and the `fill_blacklist` as well)